core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherJob.java

```diff
@@ -57,36 +57,35 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
         Stack stack = stackService.getByIdWithListsInTransaction(getStackId());
         Status stackStatus = stack.getStatus();
         if (!Status.getUnschedulableStatuses().contains(stackStatus)) {
-            applyStackPatches(stack);
+            StackPatchType stackPatchType = StackPatchType.valueOf(context.getJobDetail().getJobDataMap().getString(STACK_PATCH_TYPE));
+            applyStackPatch(stack, stackPatchType);
         } else {
-            LOGGER.debug("Existing stack fixing will be unscheduled, because stack {} state is {}", stack.getResourceCrn(), stackStatus);
+            LOGGER.debug("Existing stack patching will be unscheduled, because stack {} state is {}", stack.getResourceCrn(), stackStatus);
         }
         jobService.unschedule(context.getJobDetail().getKey());
     }
 
-    private void applyStackPatches(Stack stack) throws JobExecutionException {
-        Map<StackPatchType, Exception> errors = new HashMap<>();
-        for (ExistingStackPatchService existingStackPatchService : existingStackPatchServices) {
-            if (!existingStackPatchService.isStackAlreadyFixed(stack)) {
-                try {
-                    applyStackPatch(stack, existingStackPatchService);
-                } catch (ExistingStackPatchApplyException e) {
-                    LOGGER.error("Failed to patch stack: {} for {}", stack.getResourceCrn(), existingStackPatchService.getStackFixType(), e);
-                    errors.put(existingStackPatchService.getStackFixType(), e);
-                }
-            } else {
-                LOGGER.debug("Stack {} was already patched for {}", stack.getResourceCrn(), existingStackPatchService.getStackFixType());
+    private void applyStackPatch(Stack stack, StackPatchType stackPatchType) throws JobExecutionException {
+        ExistingStackPatchService existingStackPatchService = existingStackPatchServices.stream()
+                .filter(s -> s.getStackPatchType().equals(stackPatchType))
```

Comment:
Can s.getStackPatchType() be null? It would be safer (as the stackPatchType parameter can not be null) to swap the order of parameters to compare: stackPatchType.equals(s.getStackPatchType())

Reply:
Fixed, thanks